numcodecs/vlen.pyx

```diff
@@ -268,7 +271,7 @@ class VLenBytes(Codec):
             l = lengths[i]
             store_le32(<uint8_t*>data, l)
             data += HEADER_LENGTH
-            encv = PyBytes_AS_STRING(values[i])
```

Comment:
Looks like this line was the actual one that failed ( a bit further in than the aforementioned error would suggest: https://github.com/zarr-developers/numcodecs/issues/683#issuecomment-2791172032 )

The issue is that `value[i]` does not contain the normalized value. So if `None` shows up here, it errors. Also the error is a nasty segfault (due to attempting to access a field that does not exist) or an `assert`ion error if Python's debug mode is used